packages/carbon-graphs/src/js/controls/PairedResult/PairedResult.js

```diff
@@ -78,6 +78,24 @@ const loadInput = (inputJSON) => new PairedResultConfig()
   .clone()
   .getConfig();
 
+/**
+ * Filters invalid data like null and blank.
+ *
+ * @private
+ * @param {object} data - Data points object
+ * @returns {object} filteredData - filtered data object
+ */
+const filterPairedResultData = (data) => {
+  data.map((value) => {
+    let filteredValue = {};
+    iterateOnPairType((t) => {
+      if(!(value[t] != null && !(typeof value[t] === 'object' && Object.keys(value[t]).length === 0))){
+        delete value[t];
```

Comment:
I have some concerns about with this change.  In the previous implementation you were reconstructing the objects into a **new array** and preventing invalid values from being added.  Now with this change you are deleting properties from the objects in the array which was passed as the argument and then returning the modified array.

Since arrays and objects are passed by reference, any other place that is referencing/using this array that was passed into this function will be affected by the changes made here.  I am not a fan of mutating arguments that are passed into a function.  If you are seeing advantages to using `delete` over the previous implementation, I would prefer clone the array first and then make modifications on the cloned version and return it instead of modifying and returning the argument that was passed in.

Reply:
Fixed [here](https://github.com/cerner/terra-graphs/pull/19/commits/be9685225d23079911dd96b1245cb0ba1b5cea0f#diff-1a3e9e5e1951c8c06a171883626624b21dfb2e6131ec69031c9cf36f5e341f36R89).